client/src/app/pages/login-signup/login-signup.component.ts

```diff
@@ -1,13 +1,81 @@
 import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { User } from 'src/app/interfaces/user';
+import { StoreFacadeService } from '../../store/store-facade.service';
 
 @Component({
   selector: 'app-login-signup',
   templateUrl: './login-signup.component.html',
-  styleUrls: ['./login-signup.component.scss']
+  styleUrls: ['./login-signup.component.scss'],
 })
 export class LoginSignupComponent implements OnInit {
-  logo = '../../assets/logo.svg';
-  constructor() { }
+  hasAlpacaCode = false;
+  hasAppError = this.storeFacade.appError$;
+
+  constructor(
+    private storeFacade: StoreFacadeService,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {}
+
   ngOnInit(): void {
+    this.storeFacade.shouldAlpacaRedirect$.subscribe(
+      (shouldRedirect: boolean) => {
+        this.handleAlpacaRedirectResponse(shouldRedirect);
+      }
+    );
+
+    this.storeFacade.isUserLoggedIn$.subscribe((isUserLoggedIn: boolean) => {
+      this.handleLoginRedirect(isUserLoggedIn);
+    });
+
+    this.route.queryParams.subscribe((routeData) => {
+      this.handleRouteData(routeData);
+    });
+  }
+
+  handleAlpacaRedirectResponse(shouldRedirect: boolean): void {
+    if (shouldRedirect && !this.hasAlpacaCode) {
+      this.registerWithAlpaca();
+    }
+  }
+
+  handleLoginRedirect(isUserLoggedIn: boolean): void {
+    if (isUserLoggedIn) {
+      this.router.navigate(['/home']);
+    }
+  }
+
+  handleRouteData(routeData): void {
+    const alpacaCode = routeData ? routeData.code : null;
+    this.hasAlpacaCode = false;
+    if (!!alpacaCode) {
+      this.hasAlpacaCode = true;
+      const userParams: User = {
+        id: Number(localStorage.getItem('userId')),
+        alpacaApiKey: alpacaCode,
+      };
+      this.storeFacade.updateUser(userParams);
+    }
+  }
+
+  registerWithAlpaca(): void {
+    // TODO: This values need to be hidden / updated
+    const redirectUri = 'http://localhost:4200/'; // EnvironmentVariables.ALPACA_REDIRECT_URL;
+    const clientId = '198903d0d2f523a25e4dce65837bbf0d'; // EnvironmentVariables.ALPACA_CLIENT_ID;
```

Comment:
These need to be environment variables

Reply:
The redirect URI was fixed, but there is no point in hiding the client id as it will be exposed regardless